ethcore/src/engines/block_reward.rs

```diff
@@ -21,80 +21,104 @@ use ethabi;
 use ethabi::ParamType;
 use ethereum_types::{H160, Address, U256};
 
+use std::sync::Arc;
+use hash::keccak;
 use error::Error;
 use machine::WithRewards;
 use parity_machine::{Machine, WithBalances};
 use trace;
-use super::SystemCall;
+use super::{SystemOrCodeCall, SystemOrCodeCallKind};
 
 use_contract!(block_reward_contract, "BlockReward", "res/contracts/block_reward.json");
 
 /// The kind of block reward.
 /// Depending on the consensus engine the allocated block reward might have
 /// different semantics which could lead e.g. to different reward values.
-#[repr(u8)]
 #[derive(PartialEq, Eq, Clone, Copy, Debug)]
 pub enum RewardKind {
 	/// Reward attributed to the block author.
-	Author = 0,
-	/// Reward attributed to the block uncle(s).
-	Uncle = 1,
+	Author,
 	/// Reward attributed to the author(s) of empty step(s) included in the block (AuthorityRound engine).
-	EmptyStep = 2,
+	EmptyStep,
 	/// Reward attributed by an external protocol (e.g. block reward contract).
-	External = 3,
+	External,
+
+	/// Reward attributed to the block uncle(s) without any reference of block difference. This is used by all except Ethash engine's reward contract.
+	Uncle,
```

Comment:
I think it's safe to remove this since the block reward contract was only supported in Aura and it doesn't produce uncles, so I don't think anyone is relying on this. Even if their contract code did handle the uncle type, it would never be reached.